### PR TITLE
store/tikv: use []byte instead of kv.Key in package store/tikv

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: AndreMouche

### What problem does this PR solve?

This PR tries to use []byte instead of kv.Key in package store/tikv
Part of https://github.com/pingcap/tidb/issues/22513

### Check List 
Tests 
- Unit test

### Release note 
- No release note

### CODE REVIEW

File: store/tikv/region_cache.go [hunk 1]
```diff
@@ -592,8 +591,13 @@ func (c *RegionCache) GetTiFlashRPCContext(bo *Backoffer, id RegionVerID, loadBa
...
 type KeyLocation struct {
 	Region   RegionVerID
-	StartKey tidbkv.Key
-	EndKey   tidbkv.Key
+	StartKey []byte
+	EndKey   []byte
+}
+
+// NewKeyLocation creates a KeyLocation.
+func NewKeyLocation(region RegionVerID, start, end []byte) *KeyLocation {
```
Comment (sticnarf) [inline store/tikv/region_cache.go:599]: ditto

File: ddl/backfilling.go [hunk 2]
```diff
@@ -331,15 +331,20 @@ func splitTableRanges(t table.PhysicalTable, store kv.Storage, startKey, endKey
...
 	}
 	if len(ranges) == 0 {
 		errMsg := fmt.Sprintf("cannot find region in range [%s, %s]", startKey.String(), endKey.String())
 		return nil, errors.Trace(errInvalidSplitRegionRanges.GenWithStackByArgs(errMsg))
 	}
-	return ranges, nil
+	res := make([]kv.KeyRange, len(ranges))
+	for idx, r := range ranges {
+		res[idx] = kv.KeyRange{StartKey: kv.Key(r.StartKey), EndKey: kv.Key(r.EndKey)}
+	}
```
Comment (sticnarf) [inline ddl/backfilling.go:346]: Could it be transformed with unsafe?

File: store/copr/batch_coprocessor.go [hunk 3]
```diff
@@ -436,3 +437,11 @@ func (b *batchCopIterator) sendToRespCh(resp *batchCopResponse) (exit bool) {
 	}
 	return
 }
+
+func toTiKVKeyRanges(ranges []kv.KeyRange) *tikv.KeyRanges {
```
Comment (sticnarf) [inline store/copr/batch_coprocessor.go:441]: Could it also use unsafe to avoid extra costs?

File: ddl/backfilling.go (hunk 2 above)
Comment (disksing) [reply ddl/backfilling.go:346]: +1

**Review by disksing — APPROVED** [review]

**Review by disksing — APPROVED** [review]

File: store/driver/txn/unionstore_driver.go [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+import (
+	"context"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/kv"
+	tikvstore "github.com/pingcap/tidb/store/tikv/kv"
+	"github.com/pingcap/tidb/store/tikv/unionstore"
+)
+
+// tikvBatchGetter wrapps unionstore.BufferBatchGetter as kv.BatchGetter.
```
Comment (sticnarf) [inline store/driver/txn/unionstore_driver.go:25]:
```suggestion
// tikvBatchGetter wraps unionstore.BufferBatchGetter as kv.BatchGetter.
```

And for all same comments in the same file..

**Review by sticnarf — APPROVED** [review]
lgtm

**Pull request merged**